Sources/NIO/Codec.swift

```diff
@@ -322,22 +364,26 @@ extension ByteToMessageHandler {
     }
 
     private func processLeftovers(ctx: ChannelHandlerContext) {
-        guard self.state == .active else {
+        guard self.state.isActive else {
             // we are processing or have already processed the leftovers
             return
         }
 
-        ctx.withThrowingToFireErrorAndClose {
+        do {
```

Comment:
This change is potentially going to cause some breakage for folks, as there is now nothing that closes a HTTP connection on error by default. Same for any other B2MD.

Reply:
yes, that was deliberate. I added it now to the public API changes. Good enough?

Reply:
#600 also asks for that